<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Cytoscape StringApp</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">
                
<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section>
<h1>Cytoscape StringApp</h1>
<p style="text-align: left">In these exercises, we will use the <a
href="http://apps.cytoscape.org/apps/stringApp" target="_blank">stringApp</a> for
<a href="http://cytoscape.org/" target="_blank">Cytoscape</a> to retrieve molecular
networks from the <a href="https://string-db.org/" target="_blank">STRING</a> and
<a href="http://stitch-db.org/" target="_blank">STITCH</a> databases. The exercises
will teach you how to:</p>

<ul>
  <li>retrieve networks for proteins or small-molecule compounds of interest</li>
  <li>retrieve networks for a disease or an arbitrary topics in PubMed</li>
  <li>layout and visually style the resulting networks</li>
  <li>import external data and map them onto a network</li>
  <li>perform enrichment analyses and visualize the results</li>
  <li>merge and compare networks</li>
  <li>select proteins by attributes</li>
  <li>identify functional modules through network clustering</li>
</ul>
<br/><br/><br/><br/><p style="font-size: 60%">The original version of this tutorial was developed by Lars Juhl Jensen of the Novo Nordisk
Center for Protein Research at the University of Copenhagen.  We thank professor Jensen for his gracious willingness
to allow us to repackage the content for delivery as a Cytoscape tutorial.</p>
<div class="footer" style="position: absolute;left: -3em;bottom: -5em; ">
<img class="plain" src="StringAppLogo.png" style="height:70px;opacity:0.5"> 
<version>1.5.1</version>
</div>
</section>


<section>
<h1>Prerequisites</h1>
<p style="text-align: left">To follow the exercises, make sure that you have the
latest version of Cytoscape installed. The exercises also require you to have certain Cytoscape apps installed; <b>stringApp</b>,
<b>enhancedGraphics</b> and <b>clusterMaker2</b>, as well as the <b>yFiles</b> layout algorithms. 
<ul>
<li>Go to <highlight>Apps ‚Üí App Store ‚Üí Show App Store</highlight>, and using the search field at the top search for one of the apps, for example <b>stringApp</b>.</li>
<li>The corresponding page at the <a href="https://apps.cytoscape.org/" target="_blank">Cytoscape App Store</a> will open in a browser, click the <b>Install</b> button.</li>
<li>In the Cytoscape interface, the app should be shown with a check mark in the <highlight>App Store</highlight> in the <highlight>Control Panel</highlight>.</li>
<li>Repeat with the other apps: <b>enhancedGraphics</b>, <b>clusterMaker2</b> and <b>yFiles</b> layout algorithms.</li>
</ul> 

<p style="text-align: left">If you are not already familiar with the STRING database, we highly recommend that 
you go through the short <a href="https://jensenlab.org/training/string/" target="_blank">STRING exercises</a> 
provided by the <a href="https://jensenlab.org/" target="_blank">Jensen lab</a> to learn about the 
underlying data before working with them in these exercises.</p>
</section>


<section>
<section>
<h1 id="exercise-1">Exercise 1</h1>

<p>In this exercise, we will perform some simple queries to retrieve
molecular networks based on a protein, a small-molecule compound,
a disease, and a topic in PubMed.</p> 
</section>

<section>

<h1 id="11-protein-queries">1.1 Protein queries</h1>

<ol>
<li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING protein query</b> in the drop-down and type in a protein name, for example <b>SORCS2</b>. Alternatively, use <highlight>File &rarr; Import &rarr; Network from Public Databases...</highlight>.</li>
<li>Click the <highlight>More Options...</highlight> button ‚ò∞ to view settings. Make sure the appropriate organism is selected (e.g. Homo sapiens).</li>
<li><highlight>Maximum number of interactors</highlight> determines how
many interaction partners of your protein(s) of interest will be added to the network. The default setting is 10, which we will keep for this example.</li>
<li>Click the üîç button to start the search.</li>
</ol>
<p style="text-align: left">Unless the name(s) you entered give unambiguous matches, a
disambiguation dialog will be shown next. It lists all the matches
that the stringApp finds for each query term and selects the first
one for each. Select the right one(s) you meant and continue by
pressing the <highlight>Import</highlight> button.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string protein query query=SORCS2 taxonID=9606" class="btn btn-primary">Import SORCS2 network</a></p>
    </div>
</section>

<section>
<p style="text-align: left"><em>How many nodes are in the resulting network? How does this
compare to the maximum number of interactors you specified? What
types of information does the <highlight>Node Table</highlight>
provide?</em>
</p> 
<img class="plain" src="string-protein.png" style="height:500px">   
</section>

<section>
<h1 id="12-compound-queries">1.2 Compound queries</h1>
<ol>
<li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STITCH compound query</b> in the drop-down and type in your favorite compound, for example <b>imatinib</b>.</li>
<li>You can select the organism and number of additional interactors just like for the protein query above, and the disambiguation dialog also works the same way.</li>
<li>Click the üîç button to start the search.</li>
</ol>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string compound query query=imatinib taxonID=9606" class="btn  btn-primary">Import imatinib network</a></p>
</div>
</section>
    
<section>
<p style="text-align: left"><em>How is this network different from the protein-only network
with respect to node types and the information provided in the
<highlight>Node Table</highlight>?</em></p> 
    
<img class="plain" src="stitch-compound.png" style="height:550px">   
</section>
    
<section>
<h1 id="13-disease-queries">1.3 Disease queries</h1>
<ol>
<li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING
disease query</b> in the drop-down and type in a disease of interest, for example <b>Alzheimer‚Äôs disease</b>. The stringApp will retrieve a STRING network for the top-N proteins (by default 100) associated with the disease.</li>
<li>Click the üîç button to start the search.</li>
<li>The next dialog shows all the matches that the stringApp finds for your disease query and selects the first one. Make sure to select the intended disease before pressing the <highlight>Import</highlight> button to continue.</li>
</ol>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string disease query disease=DOID:10652 taxonID=9606 limit=100" class="btn  btn-primary">Import Alzheimer‚Äôs disease network</a></p>
</div>
</section>
    
<section>
    <p style="text-align: left"><em>Which additional attribute column do you get in the <highlight>Node Table</highlight> for a disease query compared to a protein query? (Hint: check the last column.)</em></p>
    
    <img class="plain" src="string-disease.png" style="height:600px">   

</section>

<section>
<h1 id="14-pubmed-queries">1.4 PubMed queries</h1>
<ol>
<li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING
PubMed query</b> in the drop-down and type in type a query representing a topic or interest, for example <b>jet-lag</b>. You can use any query that would work on the PubMed website, but it should obviously a topic with related genes or proteins. The stringApp will query PubMed for the abstracts, find the top-N proteins (by default 100) associated with these abstracts, and retrieve a STRING network for them.</li>
<li>Click the üîç button to start the search.</li>
<!-- <li>The next dialog shows all the matches that the stringApp finds for your disease query and selects the first one. Make sure to select the intended disease before pressing the <highlight>Import</highlight> button to continue.</li> -->
</ol>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string pubmed query pubmed=jet-lag taxonID=9606 limit=50" class="btn  btn-primary">Import jet-lag network</a></p>
</div>
</section>
    
<section>
    <p style="text-align: left"><em>Which attribute column do you get in the <highlight>Node Table</highlight> for a PubMed query compared to a disease query? (Hint: check the last columns.)</em></p>
    
    <img class="plain" src="string-pubmed.png" style="height:600px">   
    
</section>

<!--<section>
<h3 id="15-new-search-interface">1.5 New search interface</h3>

<p style="text-align: left">The types of queries described above can alternatively be performed
through the new Cytoscape <highlight>Network Search</highlight> interface at the top of 
the <highlight>Network</highlight> tab of the <highlight>Control Panel</highlight>. Click on the drop-down
menu with an icon on it. Select one of the four possible STRING queries
and directly enter your query in the text field. To change settings
such as organism, click the ‚ò∞ button next to the text field. Finally,
click the üîç button to retrieve a STRING network for your query.</p>
</section>-->
</section>

<section>
<section>

<h1 id="exercise-2">Exercise 2</h1>

<p>In this exercise, we are going to use the stringApp to query the
<a href="https://diseases.jensenlab.org" target="_blank">DISEASES</a> database for
proteins associated with epithelial ovarian cancer (EOC), retrieve a
STRING network for them, and explore the resulting network.</p>
</section>

<section>
<h1 id="21-disease-network-retrieval">2.1 Disease network retrieval</h1>
<ol>
    <li>Close the current session in Cytoscape from the menu <highlight>File ‚Üí Close Session</highlight>.</li>
    <li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING
disease query</b> in the drop-down and type in <b>ovary epithelial cancer</b>.</li>
    <li>Set the <highlight>Maximum number of proteins</highlight> option to <b>250</b>.</li>
    <li>Click the üîç button to start the search.</li>
    <li>Once the network appears, go to the menu <highlight>View ‚Üí Always Show Graphics Details</highlight> to see the individual nodes and edges.</li>
</ol>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string disease query disease=DOID:2152 taxonID=9606
limit=250" class="btn btn-primary">Import EOC disease
network</a></p>
</div>
</section>

<section>
<h1 id="22-work-with-node-attributes">2.2 Work with node attributes</h1>

<p style="text-align: left">Note that the retrieved network contains a lot of additional
information associated with the nodes and edges, such as the protein
sequence, tissue expression data, subcellular localization, disease score
(<highlight>Node Table</highlight>) as well as the confidence scores for the
different interaction evidences (<highlight>Edge Table</highlight>). In the
following, we will explore these data using Cytoscape.</p>

<ol>
   <li>Find the <highlight>disease score</highlight> column in the node attributes table (look at the last columns). Sort it by values to see the highest and lowest disease scores.</li> 
   <li>Highlight the corresponding nodes by selecting rows in the table, bringing up the context menu (right-click
       the selected rows) and choosing the <highlight>Select nodes from selected rows</highlight> option. Use one of the icons in the menu to zoom into the selected node.</li>
</ol>

<p style="text-align: left"><em>Give an example for a node with the highest and lowest disease score.</em></p>
</section>

<section>
<h1 id="23-inspect-subcellular-localization-data">2.3 Inspect subcellular localization data</h1>

<p style="text-align: left">The stringApp automatically retrieves information about
in which compartments the proteins are located from the <a
href="https://compartments.jensenlab.org/" target="_blank">COMPARTMENTS</a> database,
which we will take a look at first to better understand the data.</p>

<ol>
<li>Go to <a href="https://compartments.jensenlab.org/" target="_blank">COMPARTMENTS</a> and enter <b>ARID1A</b> into the search box. The resulting page will show all matches for the query ARID1A.</li>
<li>After selecting the human gene, you will see a schematic of where in the cell it is located and below it tables containing the specific lines of evidence that contribute to the overall score.</li>
</ol>

<p style="text-align: left"><em>What compartments is ARID1A present in with a confidence
of 5? What source do these interactions come from? Hint: you can
see what the abbreviations for different evidence types mean <a
href="http://geneontology.org/docs/guide-go-evidence-codes/" target="_blank">here</a>.</em></p>
</section>
    
<section>
<h1 id="24-continuous-color-mapping">2.4 Continuous color mapping</h1>

<p style="text-align: left">Cytoscape allows you to map attributes of the nodes and edges to
visual properties such as node color and edge width. Here, we will map
the subcellular localization data for nucleus to the node color.</p>

<ol>
<li>From the <highlight>Control Panel</highlight>, select <highlight>Style</highlight>. Click
on the <highlight>‚óÄ</highlight> button to the right of the property you want to change, in this case <highlight>Fill Color</highlight> and set <highlight>Column</highlight> to the node column containing the data that you want to use (<highlight>nucleus</highlight>).</li>
<li>Since this is a numeric value, we will use the <highlight>Continuous Mapping</highlight> as the <highlight>Mapping
Type</highlight>, and set a color gradient for how likely each protein is located in the nucleus. The default Cytoscape yellow‚Äìpurple color gradient already gives a nice visualization of the confidence of being located in this compartment.</li>
</ol>

<p style="text-align: left"><em>Many proteins are strongly associated with the nucleus ‚Äì
they will be purple.</em></p>
</section>

<section>
<h1 id="25-select-proteins-located-in-the-nucleus">2.5 Select proteins located in the nucleus</h1>

<p style="text-align: left">Because many proteins are located in the nucleus, we will identify the
proteins with highest confidence of 5. One way to do this is to use the
COMPARTMENTS sliders in the <highlight>STRING Results</highlight> panel on the
right side. 
<ol>
<li>Go to the <highlight>Node</highlight> tab and expand the group of <highlight>Compartment filters</highlight> by clicking the small triangle.</li>
<li>To hide all nodes with a compartments score below 5, find the slider for <highlight>nucleus</highlight>
and set the low bound to 5 by entering the number.</li>
</ol>
</section>
    
<section>
<p style="text-align: left"><em>How many proteins are found in the nucleus with a confidence of
5? And in mitochondrion? Hint: You can see the number of hidden nodes
in the light grey panel bar on the bottom-right part of the network view
panel, just above the Table panel.</em></p>
<img class="plain" src="nucleus-sub.png" style="height:600px">   

<p style="text-align: left"><b>Important: Move the filter back to 0 before continuing with the next exercise.</b></p>
</section>
</section>

<section>
<section>
<h1 id="exercise-3">Exercise 3</h1>

<p>In this exercise, we will work with a list of 541
proteins associated with epithelial ovarian cancer (EOC)
as identified by phosphoproteomics in the study by <a
href="https://doi.org/10.1016/j.celrep.2017.03.015">Francavilla et
al.</a>. An adapted, simplified version of their results table can be
downloaded <a href="https://raw.githubusercontent.com/cytoscape/cytoscape-automation/master/for-scripters/R/notebooks/stringApp/Francavilla2017CellRep.tsv" target="_blank">here</a>. Download the file, and open it in Excel or a similar tool.</p>
</section>

<section>
<h1 id="31-protein-network-retrieval">3.1 Protein network retrieval</h1>

<ol>
    <li>In the <highlight>Network tab</highlight> of the <highlight>Contol Panel</highlight>, select <b>STRING
protein query</b> in the drop-down and paste the list of UniProt accession numbers from the UniProt column in the table.</li>
    <li>Leave the default value for <highlight>Maximum number of interactor</highlight>.</li>
    <li>The disambiguation dialog shows all STRING proteins that cannot be matched to the query terms uniquely, with the first protein for each query term automatically selected. This default is fine for this exercise</li>
    <li>Click the üîç button to start the search.</li>
    <li>Once the network appears, go to the menu <highlight>View ‚Üí Always Show Graphics Details</highlight> to see the individual nodes and edges.</li>
</ol>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string protein query query=&quot;A6ND36,A6NKT7,B1AJR3,B4DY08,B4E106,O00161,O00192,O00264,O00499,O00767,O14578,O14607,O14647,O14681,O14683,O14745,O14828,O14924,O14974,O14979,O15014,O15027,O15164,O15231,O15320,O15439,O43166,O43172,O43290,O43395,O43399,O43491,O43719,O43731,O60551,O60716,O60784,O60841,O60885,O75113,O75179,O75351,O75369,O75376,O75400,O75427,O75448,O75569,O75643,O75909,O75925,O75970,O76021,O94979,O95049,O95071,O95171,O95218,O95248,O95249,O95365,O95391,O95400,O95453,O95674,O95772,O95785,O95831,O96013,O96017,O96019,P00533,P02538,P02545,P02686,P02765,P04920,P05386,P05412,P05783,P06748,P07199,P07910,P08238,P08559,P08670,P08727,P08729,P09601,P09651,P11137,P11388,P12694,P12931,P12956,P13639,P13647,P14923,P15056,P15336,P15408,P15923,P16104,P16949,P17096,P17181,P17480,P17844,P17980,P18583,P18615,P18669,P18887,P19174,P19338,P19634,P20042,P20700,P20810,P21127,P21333,P21796,P22087,P22626,P23193,P23246,P23588,P24928,P25205,P25445,P26368,P26373,P27816,P27824,P29590,P29692,P30414,P30622,P31930,P31943,P31949,P35269,P35579,P35611,P35613,P35658,P36578,P40123,P41212,P42166,P42696,P43243,P45880,P46821,P48634,P48681,P48960,P49207,P49321,P49585,P49736,P49756,P49757,P49792,P50613,P51608,P52594,P55287,P55884,P56181,P58107,P61247,P61313,P62328,P62753,P62979,P62995,P68363,P78332,P78527,P80723,P84103,P84157,P85037,P98175,Q00536,Q00839,Q01082,Q01105,Q01130,Q01167,Q02078,Q02241,Q02543,Q02818,Q02880,Q02952,Q03135,Q03164,Q04637,Q05397,Q05519,Q05655,Q05682,Q07157,Q07955,Q08945,Q09666,Q12769,Q12830,Q12872,Q12888,Q12906,Q12996,Q13112,Q13153,Q13177,Q13185,Q13242,Q13243,Q13247,Q13263,Q13286,Q13425,Q13427,Q13428,Q13443,Q13451,Q13459,Q13523,Q13796,Q13838,Q13946,Q14004,Q14008,Q14151,Q14156,Q14160,Q14207,Q14289,Q14498,Q14573,Q14669,Q14684,Q14694,Q14739,Q14764,Q14814,Q14865,Q14966,Q14978,Q14980,Q15019,Q15059,Q15061,Q15149,Q15154,Q15233,Q15390,Q15424,Q15459,Q15477,Q15678,Q16181,Q16204,Q16630,Q16647,Q16666,Q24JP5,Q27J81,Q2LD37,Q2M2I8,Q3KQU3,Q53EL6,Q53F19,Q53GS9,Q5HYI7,Q5JRA6,Q5JSH3,Q5JSZ5,Q5JTV8,Q5JVS0,Q5SNT6,Q5SSJ5,Q5SW79,Q5SYE7,Q5T200,Q5T4S7,Q5T5C0,Q5TAQ9,Q5TCQ9,Q5TCZ1,Q5VT52,Q5VTL8,Q5VUA4,Q5VZK9,Q5VZL5,Q658Y4,Q66K74,Q69YN4,Q6DD88,Q6KC79,Q6NYC8,Q6P4E1,Q6PD62,Q6PKG0,Q6WCQ1,Q6Y7W6,Q6ZRV2,Q71RC2,Q76M96,Q7KYR7,Q7KZ85,Q7KZF4,Q7L1V2,Q7L4E1,Q7L4I2,Q7L8J4,Q7Z3K3,Q7Z417,Q7Z460,Q7Z478,Q7Z4V5,Q7Z5L9,Q7Z6Z7,Q86UE4,Q86UP2,Q86VM9,Q86VR2,Q86W92,Q8IU81,Q8IVF2,Q8IWS0,Q8IX90,Q8IXQ4,Q8IYB3,Q8IYI6,Q8N201,Q8N3V7,Q8N4C8,Q8N554,Q8N684,Q8N720,Q8N8A6,Q8N8S7,Q8N9T8,Q8NB78,Q8NC51,Q8ND56,Q8ND76,Q8NDX5,Q8NEY1,Q8NHG8,Q8TAQ2,Q8TBE7,Q8TDB6,Q8TDM6,Q8TDY2,Q8TEB9,Q8TEU7,Q8WU90,Q8WUB8,Q8WUI4,Q8WUM9,Q8WUY3,Q8WVC0,Q8WVM0,Q8WWI1,Q8WWM7,Q8WWQ0,Q8WX93,Q8WY36,Q8WYL5,Q8WZ74,Q92466,Q92508,Q92522,Q92538,Q92543,Q92545,Q92585,Q92597,Q92599,Q92610,Q92922,Q92945,Q92974,Q969E4,Q96AQ6,Q96B36,Q96E09,Q96E39,Q96EB6,Q96EV2,Q96F86,Q96FV9,Q96G23,Q96GX8,Q96I25,Q96J92,Q96JM3,Q96L91,Q96PK6,Q96PY5,Q96QR8,Q96S55,Q96S94,Q96ST2,Q96ST3,Q96T23,Q96T37,Q96T58,Q96TA1,Q99442,Q99567,Q99569,Q99590,Q99700,Q99959,Q9BQQ3,Q9BRD0,Q9BRJ6,Q9BSQ5,Q9BST9,Q9BTA9,Q9BTX1,Q9BU76,Q9BW71,Q9BWF3,Q9BX95,Q9BXF6,Q9BXK5,Q9BXP5,Q9BYW2,Q9BZ71,Q9BZZ5,Q9C0C2,Q9GZY8,Q9H0C8,Q9H0D6,Q9H1E3,Q9H2H9,Q9H2U2,Q9H307,Q9H3R2,Q9H4L5,Q9H5H4,Q9H6F5,Q9H7F0,Q9H8Y8,Q9HAU0,Q9HB09,Q9HCD5,Q9HCG8,Q9HCM4,Q9NPH3,Q9NQC3,Q9NQG5,Q9NQG6,Q9NRA0,Q9NRA8,Q9NRG0,Q9NTI5,Q9NVM6,Q9NWB6,Q9NWV8,Q9NWW5,Q9NX40,Q9NX94,Q9NXD2,Q9NY27,Q9NY61,Q9NYF8,Q9NYM9,Q9NYV4,Q9NZ32,Q9NZJ7,Q9NZT2,Q9P0B6,Q9P0K7,Q9P1Y6,Q9P206,Q9P227,Q9P2E9,Q9P2N5,Q9P2R6,Q9UBC2,Q9UBI6,Q9UBV2,Q9UDY2,Q9UEY8,Q9UGU0,Q9UH62,Q9UHB6,Q9UHB7,Q9UHD8,Q9UHR4,Q9UHW9,Q9UID3,Q9UIG0,Q9UK61,Q9UKA4,Q9UKK3,Q9UKV3,Q9ULH0,Q9ULT0,Q9UMZ2,Q9UN86,Q9UNF1,Q9UNL2,Q9UP95,Q9UPN3,Q9UPU5,Q9UPU7,Q9UQ35,Q9UQE7,Q9Y222,Q9Y2D5,Q9Y2J2,Q9Y2K6,Q9Y2V2,Q9Y2W1,Q9Y2W2,Q9Y320,Q9Y385,Q9Y3T9,Q9Y490,Q9Y4G6,Q9Y4J8,Q9Y519,Q9Y5J1,Q9Y5Q9,Q9Y5S9,Q9Y618,Q9Y6D6,Q9Y6G9,Q9Y6I3,Q9Y6K1,Q9Y6M7,Q9Y6X9,Q9Y6Y8&quot; taxonID=9606 limit=0" class="btn btn-primary">Import network</a></p>
</div>
</section>

<section>
<p style="text-align: left"><em>How many nodes and edges are there in the resulting network? Do
    the proteins all form a connected network? Why?</em></p>

<img class="plain" src="exercise3-net.png" style="height:600px">

</section>
    
<section>
    <p style="text-align: left">Cytoscape provides several visualization options under the
<highlight>Layout</highlight> menu. Try the <highlight>Degree Sorted Circle
Layout</highlight>, the <highlight>Prefuse Force Directed Layout</highlight>
with score as edge weight, and <highlight>yFiles Organic Layout</highlight>.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:layout kamada-kawai" class="btn btn-primary">Layout network</a></p>
</div>

<p style="text-align: left"><em>Can you find a layout that allows you to easily recognize patterns in the network? What about the Edge-weighted Spring Embedded Layout with the attribute ‚Äòscore‚Äô, which is the combined STRING interaction score?</em></p>

<div class="fragment">
<img class="plain" src="exercise3-net-layout.png" style="height:570px"></div>
    
</section>
    
<section>
<h1 id="32-discrete-color-mapping">3.2 Discrete color mapping</h1>

<p style="text-align: left">Cytoscape allows you to map attributes of the nodes
and edges to visual properties such as node color and edge
width. Here, we will map drug target family data from the <a
href="https://pharos.nih.gov/idg/targets" target="_blank">Pharos</a> database to the node
color. This data is contained in the node attribute called <highlight>target
family</highlight>.</p>

<ol>
    <li>Select <highlight>Style</highlight> from the <highlight>Control Panel</highlight>.</li> 
    <li>Click the <highlight>‚óÄ</highlight> button to the right of the property you
want to change, in this case <highlight>Node Fill Color</highlight>, and change
<highlight>Column</highlight> from <b>name</b> to <b>family</b>, which is the node column
    containing the data that you want to use. </li>
<p style="text-align: left">This action will remove the
rainbow coloring of the nodes and present you with a list of all the
different values of the attributes that exist in the network.</p>
</ol>
<p style="text-align: left"><em>Which target families are present in the network?</em></p>
</section>

<section>
<ol>
<li>To color the corresponding proteins, first click the field
to the right of an attribute value, i.e. <highlight>GPCR</highlight> or
<highlight>Kinase</highlight>, then click the ‚ãØ button and choose a color from the color selection dialog.</li>
<li>You can also set a default color, e.g. for all nodes that do not have a target family annotation from Pharos, by clicking on the grey button in the first column of the same row.</li>
</ol>
<p style="text-align: left"><em>How many of the proteins in the network are ion channels or GPCRs?</em></p>

<img class="plain" src="exercise3-net-families.png" style="height:550px">

</section>
    
<section>
    
<p style="text-align: left">There are many kinases in the network. We can avoid counting them manually by creating a selection filter.</p>
<ol>
<li>Click on the <highlight>Filter</highlight> tab in the <highlight>Control Panel</highlight>.</li>
<li>Click the <highlight>·ê©</highlight> button and choose <highlight>Column filter</highlight> from the drop-down menu. Then, find and select the attribute <highlight>Node: family</highlight>. Write <highlight>kinase</highlight> in the text field to select all nodes with this annotation.</li>
</ol>
<!-- [Create filter for kinases](cycmd:filter create name="myfilter" json='{ "id" : "ColumnFilter", "parameters" : { "criterion" : "Kinase", "columnName" : "target::family", "predicate" : "CONTAINS"} }'){: .btn .btn--cybrowser .btn--primary}
 -->

<p style="text-align: left"><em>How many kinases are in the network?</em></p>
</section>

<section>
<h1 id="33-data-import">3.3 Data import</h1>

<p style="text-align: left">Network nodes and edges can have additional information associated with
them that we can load into Cytoscape and use for visualization. We will
import the data from the <a href="https://raw.githubusercontent.com/cytoscape/cytoscape-automation/master/for-scripters/R/notebooks/stringApp/Francavilla2017CellRep.tsv" target="_blank">text
file</a>.</p>

<ol>
<li>To import the node attributes file into Cytoscape, go to <highlight>File
‚Üí Import ‚Üí Table from File</highlight>. The preview in the import
dialog will show how the file is interpreted given the current settings
    and will update automatically when you change them.</li> 
<li>To change the default selection, click the arrow in the column heading. For example,
you can decide whether the column is imported or not by changing the
<highlight>Meaning</highlight> of the column (hover over each symbol with
the mouse to see what they mean). This column-specific dialog will also
allow you to change the column name and type.</li> 

</ol>
<p style="text-align: left">Now you need to map unique identifiers between the entries in the data
and the nodes in the network. The key point of this is to identify which
nodes in the network are equivalent to which entries in the table. This
enables mapping of data values into visual properties like <b>Fill Color</b>
and <b>Shape</b>. This kind of mapping is typically done by comparing the unique
identifier attribute value for each node (<highlight>Key Column for Network</highlight>) with the
unique identifier value for each data value (key symbol). As a default,
Cytoscape looks for an attribute value of ‚ÄòID‚Äô in the network and
a user-supplied <highlight>Key</highlight> in the dataset.</p>
</section>

<section>
<p style="text-align: left">The <highlight>Key Column for Network</highlight> can be changed using a
combo box and allows you to set the node attribute column that is to be
used as key to map to. 
<ol>
<li>In this case, we will use <highlight>query term</highlight> because this attribute contains the UniProt accession numbers you entered when retrieving the network.</li>
<li>You can also change the <highlight>Key</highlight> by pressing the key button for the column that is to be used as key for mapping values in the dataset. In this case it is the first column in the table called
UniProt, from where you copied the identifiers.</li>
<li>Click <highlight>OK</highlight> to import the data.</li>
</ol>
<p style="text-align: left">If there is a match between the value of a <highlight>Key</highlight> in the dataset
and the value the <highlight>Key Column for Network</highlight> field in the network, all
attribute‚Äì-value pairs associated with the element in the dataset are
assigned to the matching node in the network. You will find the imported
columns at the end of the <highlight>Node Table</highlight>.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:table import url
url=&quot;https://jensenlab.org/assets/Francavilla2017CellRep.tsv&quot;
keyColumnForMapping=&quot;query term&quot; keyColumnIndex=1
startLoadRow=1" class="btn btn-primary">Import
table</a></p>
</div>
</section>

<section>
<h1 id="34-continuous-color-mapping">3.4 Continuous color mapping</h1>

<p style="text-align: left">Now, we want to color the nodes according to the quantitative
phosphorylation data (log ratio) between disease and healthy tissues
for the most significant site for each protein. 
<ol>
<li>From the <highlight>Control Panel</highlight>, select <highlight>Style</highlight>. Then click on the <highlight>‚óÄ</highlight> button
to the right of the property you want to change, for example <highlight>Node Fill Color</highlight>.</li>
<li>Next, set <highlight>Column</highlight> to the node column containing the data that you want to use (EOC vs EOS&amp;FTE).</li> <li>Since this is a numeric value, we will use the <highlight>Continuous Mapping</highlight> as the <highlight>Mapping Type</highlight>, and set a color gradient for how abundant each protein is. The default Cytoscape color gradient
blue‚Äìwhite‚Äìred already gives a nice visualization of the log
ratio.</li>
</ol>
</section>
<section>
<p style="text-align: left"><em>Are the up-regulated nodes grouped together? Do you see any issues
with the color gradient?</em></p>

<ol>
<li>To change the colors, double-click on the color gradient in order to
bring up the <highlight>Continuous Mapping Editor</highlight> window and edit
the colors for the continuous mapping.</li> 
<li>In the mapping editor dialog, the color that will be used for the minimum value is on the left, and
the maximum is on the right. Double-click on the triangles on the top
and sides of the gradient to change the colors.</li>
<li>The triangles on the top
represent the values at which the data will be clipped; anything above
the right triangle will be set to the max value. This is useful if you
have a small number of values that are significantly higher than the
median. As you move the triangles and change the color, the display in
the network pane will automatically update ‚Äì this is all easier to do
than to explain!</li> 
<li>If at any point it does not seem to work as expected,
it is easiest to just delete the mapping and start again.</li>
</ol>

<p style="text-align: left"><em>Can you improve the color mapping such that it is easier to see
which nodes have a log ratio below -4 and above 4?</em></p>
</section>

<section>
<h1 id="35-network-clustering">3.5 Network clustering</h1>

<p style="text-align: left">Next, we will use the MCL algorithm to identify clusters of
tightly connected proteins within the network. Go to the menu
<highlight>Apps ‚Üí clusterMaker Cluster Network ‚Üí MCL Cluster</highlight>. Set the
<highlight>Granularity parameter (inflation value)</highlight> to 4 and choose
the <highlight>stringdb::score</highlight> attribute (i.e. the overall STRING
confidence score) as <highlight>Array Source</highlight>, select the option
<highlight>Create new clustered network</highlight>, and click OK to start the
clustering. The app will now run the algorithm and automatically create
a network showing the clusters.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:cluster mcl attribute=score inflation_parameter=4
showUI=true" class="btn btn-primary">Cluster
network</a></p>
</div>

<p style="text-align: left"><em>How many clusters have at least 10 nodes?</em></p>

<p style="text-align: left">We will work with the largest cluster in the network (it should be
in the upper left corner). Select the nodes of this cluster by holding
down the modifier key (Shift on Windows, Ctrl or Command on Mac) and
then left-clicking and dragging to select multiple nodes. Then, create
a new network by clicking on the <highlight>New Network from Selection
(All Edges)</highlight> <img class="plain" src="NewNetworkFromSelectionButton.png"> button.</p>

<!-- [Select cluster](cycmd:filter create name="myfiltercluster1" json='{ "id" : "ColumnFilter", "parameters" : { "criterion" : [ 1, 1 ], "columnName" : "__mclCluster", "predicate" : "BETWEEN"} }'){: .btn .btn--cybrowser .btn--primary}
[Create new network from selection](cycmd:network create excludeEdges=false nodeList=selected source=current){: .btn .btn--cybrowser .btn--primary}
 -->

<p style="text-align: left"><em>How many nodes and edges are there in this cluster?</em></p>
</section>

<section>
<h1 id="36-functional-enrichment-and-enriched-publications">3.6 Functional enrichment and enriched publications</h1>

<p style="text-align: left">Next, we will retrieve functional enrichment for the proteins in our
network of the largest cluster. 
<!-- First we will have to tell Cytoscape
that the new network we created is a STRING network, go to the menu
<highlight>Apps ‚Üí STRING ‚Üí Set as STRING network</highlight>. -->
</p>

<p style="text-align: left">After making sure that no nodes are selected in the network, go
to the menu <highlight>Apps ‚Üí STRING Enrichment ‚Üí Retrieve functional
enrichment</highlight> and keep the default settings. A new STRING Enrichment
tab will appear in the <highlight>Table Panel</highlight> on the bottom. It
contains a table of enriched terms and corresponding information for
each enrichment category. You can see which proteins are annotated with
a given term by selecting the term in the <highlight>STRING Enrichment
panel</highlight>.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string make string" class="btn btn-primary">Set as STRING network</a>
<a href="cycmd:string retrieve enrichment" class="btn btn-primary">Retrieve functional enrichment</a>
<a href="cycmd:string show enrichment" class="btn btn-primary">Show functional enrichment</a></p>
</div>

<p style="text-align: left"><em>Which are the four most statistically significant terms? Do the Uniprot and GO Process terms agree with each other, i.e., annotate the same set of nodes?</em></p>

<!-- To explore only specific types of terms, e.g. GO terms, and to remove redundant terms from the table, click on the filter icon in the **Table panel** (leftmost icon). Select the three types of GO terms, enable the option to **Remove redundant terms** and set **Redundancy cutoff** to 0.2. In this way, you will see only the statistically significant GO terms that do not represent largely the same set of proteins within the network. You can see which proteins are annotated with a given term by selecting the term in the **STRING Enrichment** panel.

[Filter functional enrichment](cycmd:string filter enrichment categories="GO Process,GO Component,GO Function" overlapCutoff=0.2 removeOverlapping=true){: .btn .btn--cybrowser .btn--primary}
 -->

<p style="text-align: left">Next, we will visualize the top-5 enriched terms in the network using
split charts, click the colorful chart icon to show the terms as the
charts on the network. You can manually change the layout of the network
to improve the visualization. First apply the <highlight>yFiles Organic
Layout</highlight> and then scale the network to reduce the overlap of
the charts using the <highlight>Node Layout Tools (Layout ‚Üí Layout
Tools</highlight>).</p>
</section>
<section>

<p style="text-align: left">To retrieve a list of publications that are enriched for the proteins
in the network, go to the menu <highlight>Apps ‚Üí STRING Enrichment
‚Üí Retrieve enriched publications</highlight>. A new <highlight>STRING
Publications</highlight> tab will appear in the <highlight>Table Panel</highlight>
on the bottom. It contains a table of enriched publications and associated
information such as how many of the network proteins were mentioned in
each publication.</p>

<div class="cybrowser">
<p style="text-align: left"><a href="cycmd:string retrieve publications" class="btn btn-primary">Retrieve enriched publications</a>
<a href="cycmd:string show publications" class="btn btn-primary">Show enriched publications</a></p>
</div>

<p style="text-align: left"><em>What is the title of the most recent publication?</em></p>

<p style="text-align: left">To save the list of enriched terms and associated p-values as a text
file, go to <highlight>Apps ‚Üí STRING Enrichment ‚Üí Export enrichment
results</highlight>.</p>
</section>

<section>
<h1 id="37-overlap-networks">3.7 Overlap networks</h1>

<p style="text-align: left">Cytoscape provides functionality to merge two or more networks,
building either their union, intersection or difference. We will now
merge the EOC network we have from the DISEASES query with the one we
have from the data, so that we can identify the overlap between them. Use
the Merge tool (<highlight>Tools ‚Üí Merge ‚Üí Networks‚Ä¶</highlight>) and
select the <highlight>Intersection</highlight> button. Then, select the two
STRING networks from <highlight>Available Networks</highlight> list (‚ÄòString
Network - ovary epithelial cancer‚Äô and ‚ÄòString Network‚Äô). Click
on <highlight>&gt;</highlight> to add them to the list of <highlight>Networks to
Merge</highlight> and click <highlight>Merge</highlight>.</p>

<p style="text-align: left"><em>How many nodes are in the intersection?</em></p>
</section>

<section>
<h1 id="38-integrate-networks">3.8 Integrate networks</h1>

<p style="text-align: left">Now we will make the union of the intersection network, which
contains the disease scores, and the experimental network. Use the
<highlight>Merge</highlight> tool again to make the <highlight>Union</highlight>
of ‚ÄòMerged Network‚Äô and ‚ÄòString Network‚Äô. Make sure that the
new merged network has the same number of nodes and edges as ‚ÄòString
Network‚Äô, and that some nodes have a disease score.</p>

<p style="text-align: left">Now, we can change the visualization of the merged network to be
able to identify high disease score proteins. Specifically, we will
change the size of the nodes in function of their disease score. Select
<highlight>Style</highlight> in the <highlight>Control Panel</highlight>. Click on the
<highlight>Lock node width and height</highlight> option to enable it so that
the nodes have only one attribute <highlight>Size</highlight> instead of two
attributes <highlight>Height</highlight> and <highlight>Width</highlight>. Click on
the ‚óÄ button to add a continuous mapping of the <highlight>Size</highlight>
attribute using the <highlight>disease score</highlight>. Modify the values
so that by default a node size is <b>30</b>; the mapping should go from <b>35
to 50</b>. To change the default value, you have to click on the default
<b>35.0</b> value at the left of the <highlight>Size</highlight> attribute. To change
the mapping values, double click on the chart and then to double click
on the square corresponding to the value you want to modify and set the
value you want (35 and 50).</p>
</section>
</section>

<section>
<section>
<h1 id="exercise-4">Exercise 4</h1>

<p>In this exercise, we will retrieve virus-host networks for two closely
related viruses, merge them into a single network, and then will retrieve
the functional enrichment for the host proteins in this network.</p>
</section>

<section>
<h1 id="41-virus-queries">4.1 Virus queries</h1>

<p style="text-align: left">Go to the menu <highlight>File ‚Üí Import ‚Üí Network from Public
Databases</highlight>. In the import dialog, choose <highlight>STRING: protein
query</highlight> as the <highlight>Data Source</highlight>. As of version 1.4 of
the STRING app, 236 virus species are included in the species dropdown
menu. Since most viruses are small (they have a median of 9 proteins in
their genomes) it is reasonable to import <highlight>all proteins of this
species</highlight> for a given virus, so select this checkbox underneath
the species dropdown. For this example we will query all proteins of
‚ÄúHuman papillomavirus type 16 (HPV 16)‚Äù. Simply type HPV 16 and
select the species from the resulting shorter dropdown menu.</p>

<p style="text-align: left"><em>How many virus proteins are encoded for by this virus? What node
information is imported along with the names of the proteins?</em></p>
</section>
<section>

<h1 id="42-expand-with-host-interactors">4.2 Expand with host interactors</h1>

<p style="text-align: left">To retrieve interactions with host proteins, go to <highlight>Apps
‚Üí STRING ‚Üí Expand network</highlight>. In the resulting dialog, enter
the number of desired host proteins, and select the host species from
<highlight>Type of interactors to expand network by</highlight>. All host
species for which we have interactions with the currently imported virus
genes, will be shown in the dropdown menu. The <highlight>selectivity of
interactors</highlight> can also be specified ‚Äì we recommend a default
value of <b>0.5</b>, but you can move the slider towards 0 to decrease the
number of network-sepcific interactors or towards 1 to increase it. In
this example, we will import <b>10</b> human proteins, and keep the default
selectivity.</p>

<p style="text-align: left">The resulting network will be automatically re-styled such that
the nodes representing virus proteins are red and host proteins are
green-blue. These attributes can be changed from the Cytoscape Style
menu.</p>

<p style="text-align: left"><em>Which human protein has the highest interaction score to one of
the virus proteins? What cellular functions is this protein involved
in? (Hint: check the STRING results panel.)</em></p>

<p style="text-align: left">Additional viruses or hosts can be added to the network by iterating
on this procedure, but this will only add proteins that interact with the
proteins that are already in the network. This will work fine when adding
new hosts, since all virus proteins are already in the network. However
to add new viruses, we recommend merging the expanded networks for
each virus.</p>
</section>
<section>

<h1 id="43-add-specific-host-proteins">4.3 Add specific host proteins</h1>

<p style="text-align: left">If a specific host protein is desired, it can also be included in
the network from the <highlight>Apps ‚Üí STRING ‚Üí Query for additional
nodes</highlight> menu option. In this example, p53 is not one of the
proteins that was included in the network in the previous step, however
it is known that the HPV E6 protein mediates ubiquitination of p53. To
include this protein, choose <b>Homo sapiens</b> for the species (you may
have to scroll up in the list), and enter <b>tp53</b> into the text area
box in the dialog, then click <highlight>Import</highlight>.</p>

<p style="text-align: left"><em>Which HPV proteins does p53 interact with?</em></p>

<p style="text-align: left">Note that p53 will be added to the network in the previous step if more
proteins are imported or the selectivity is set to a lower value. Choosing
a lower selectivity will include more hub proteins (such as p53) that are
connected to many proteins, and that do not interact specifically with
proteins in your network. Conversely, choosing a higher selectivity will
include more proteins that are more specific to your network, but these
interactions will have lower confidence (since any higher confidence
hub proteins will be filtered out). Further, be aware that changing the
selectivity parameter will change the enrichment results in step 4.5,
since different proteins will be included in the host network.</p>

</section>
<section>
<h1 id="44-merge-two-host-virus-networks">4.4 Merge two host-virus networks</h1>

<p style="text-align: left">Let us now compare the networks for HPV 16 and HPV 1a. Create a new
host-virus network for ‚ÄúHuman papillomavirus type 1a (HPV 1a)‚Äù by
repeating steps 4.1 and 4.2. Merge the two networks using <highlight>Tools
‚Üí Merge ‚Üí Networks</highlight>. Move both the HPV 16 and HPV 1a networks
into the <highlight>Networks to merge</highlight> box and otherwise use the
defaults for the merge. In the resulting network, use the menu option
<highlight>Apps ‚Üí STRING ‚Üí Set as STRING network</highlight> to manipulate
the network as a STRING network again. To show any interactions between
host nodes that were present in one source network but not the other,
first set the confidence to 1, then set the confidence to the desired
confidence (0.4) to retrieve any missing interactions.</p>

<p style="text-align: left">The resulting network can be styled to give the nodes of each
species a distinct color so that the proteins of the two viruses can be
distinguished from each other.</p>

<p style="text-align: left"><em>How many host proteins interact with E6 from both HPV species?</em></p>

</section>
<section>
<h1 id="45-functional-enrichment">4.5 Functional enrichment</h1>

<p style="text-align: left">We will now examine the human proteins to see what pathways are enriched in this network.</p>

<p style="text-align: left">Next, we will retrieve functional enrichment for the human proteins. Go
to the menu <highlight>Apps ‚Üí STRING Enrichment ‚Üí Retrieve functional
enrichment</highlight> and keep the default p-value of 0.05. Homo sapiens
will be selected by default in the species dropdown. It is currently only
possible to retrieve enrichment for host proteins. A new STRING Enrichment
tab will appear in the Table Panel. It contains a table of enriched
terms and corresponding information for each enrichment category. Use
the filter button in the top left of the STRING Enrichment panel to show
only <highlight>KEGG Pathways</highlight>. Click on the draw charts icon to the
right of the filter icon to plot the enrichment values on the network.</p>

<p style="text-align: left"><em>Which two KEGG pathways have the lowest p-values? Which host
proteins are associated with the KEGG pathways ‚Äúcell cycle‚Äù? (Hint:
click on the associated row in the enrichment table to select the proteins
with this term.)</em></p>
</section>
</section>

<section>
<h1 id="supporting-literature">Supporting literature</h1>

<p style="text-align: left">Doncheva NT, Morris JH, Gorodkin J and Jensen LJ (2018). Cytoscape stringApp: Network analysis and visualization of proteomics data.<br />
<a href="https://doi.org/10.1101/358283" target="_blank">Preprint</a></p>

<p style="text-align: left"><a href="https://creativecommons.org/licenses/by/4.0/"><img src="https://i.creativecommons.org/l/by/4.0/88x31.png" alt="CC BY 4.0" /></a></p>
        
</section>
                
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>

<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>

		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
                controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },
                
                { src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
                { src: '../../plugin/notes/notes.js', async: true },
                { src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                { src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
