<!doctype html>
<html>
	<head>
		<meta charset="utf-8">
		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

		<title>Functional Enrichment Analysis</title>

		<link rel="stylesheet" href="../../css/reveal.css">
		<link rel="stylesheet" href="../../css/theme/cytoscape-protocol-1.css">

		<!-- Theme used for syntax highlighting of code -->
		<link rel="stylesheet" href="../../lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? '../../css/print/pdf.css' : '../../css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
	</head>
	<body>
		<div class="reveal">
			<div class="slides">

<!-- ========================================================== -->
<!-- THIS IS WHERE YOUR CONTENT BEGINS -->

<section id='title'>
    <h1>Functional Enrichment Analysis</h1>
    <p>This protocol is designed as a walk-through tour of popular functional enrichment analysis tools and describes the use of three functional enrichment tools:</p>
    <ul>
        <li><a href="https://maayanlab.cloud/Enrichr/" target="_blank">Enrichr</a></li>
        <li><a href="http://www.webgestalt.org/" target="_blank">WebGestalt</a></li>
        <li><a href="https://github.com/gladstone-institutes/Interactive-Enrichment-Analysis" target="_blank">Interactive Enrichment Analysis</a></li>
    </ul>
</section>

<!--<section id='example_data_tcga'>
        <h1>Example Data - TCGA lung cancer study</h1>
        <p>This tour can be used with your own data, or with an example dataset. Three example datasets are provided in the following slides.</p>
        <p>The first dataset is from the <a href="https://www.cancer.gov/types/lung" target="_blank">TCGA lung cancer study</a>, and includes a comparison of the expression of transcripts in lung cancer biopses versus normal tissue. The data was processed to produce the example files:</p>
    <ol>
        <li class="fragment" data-fragment-index="1">Calculate <b>log2 fold change</b> and <b>adjusted pvalue</b> for cancer vs normal per gene from group average expression values.</li>
        <li class="fragment" data-fragment-index="2">Filter for significantly up-regulated genes: <b>log2 fold change => 2 AND adjusted pvalue <= 0.05</b>; <highlight>list of 367 up-regulated genes (NCBI gene symbol)</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.up.txt" target="_blank">Download data</a>.</li>
        <li class="fragment" data-fragment-index="4">Filter for significantly down-regulated genes: <b>log2 fold change <= -2 AND adjusted pvalue <= 0.05</b>; <highlight>list of 516 down-regulated genes (NCBI gene symbol)</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.dn.txt" target="_blank">Download data</a>.</li>
        <li class="fragment" data-fragment-index="6">Gene ranking calculated for all genes using <b>sign(log2 fold change) * -log10(pvalue)</b>; <highlight>file with Ensembl gene id and rank value for all genes</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.ensembl.gsea.txt" target="_blank">Download data</a>.</li>
    </ol>
    <p class="fragment" data-fragment-index="8">These input files can be used for <b>ORA</b> and <b>GSEA</b> analysis in <b>Enrichr</b> and <b>WebGestalt</b>.</p>
     <div class="fragment fade-in-then-out" data-fragment-index="3" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_up.png" style="width:70%">
     </div>
     <div class="fragment fade-in-then-out" data-fragment-index="5" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_down.png" style="width:70%">
     </div>
     <div class="fragment fade-in-then-out" data-fragment-index="7" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_ranked.png">
     </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>-->
                
<section id='example_data_tcga'>
        <h1>Example Data - TCGA lung cancer study</h1>
        <p>This tour can be used with your own data, or with an example dataset. Three example datasets are provided in the following slides.</p>
        <p>The first dataset is from the <a href="https://www.cancer.gov/types/lung" target="_blank">TCGA lung cancer study</a>, and includes a comparison of the expression of transcripts in lung cancer biopses versus normal tissue. The data was processed to produce the example files:</p>
    <ol>
        <li>Calculate <b>log2 fold change</b> and <b>adjusted pvalue</b> for cancer vs normal per gene from group average expression values.</li>
        <li>Filter for significantly up-regulated genes: <b>log2 fold change => 2 AND adjusted pvalue <= 0.05</b>; <highlight>list of 367 up-regulated genes (NCBI gene symbol)</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.up.txt" target="_blank">Download data</a>.</li>
        <li>Filter for significantly down-regulated genes: <b>log2 fold change <= -2 AND adjusted pvalue <= 0.05</b>; <highlight>list of 516 down-regulated genes (NCBI gene symbol)</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.dn.txt" target="_blank">Download data</a>.</li>
        <li>Gene ranking calculated for all genes using <b>sign(log2 fold change) * -log10(pvalue)</b>; <highlight>file with Ensembl gene id and rank value for all genes</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.ensembl.gsea.txt" target="_blank">Download data</a>.</li>
    </ol>
    <p>These input files can be used for <b>ORA</b> and <b>GSEA</b> analysis in <b>Enrichr</b> and <b>WebGestalt</b>.</p>
     <!--<div class="fragment fade-in-then-out" data-fragment-index="3" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_up.png" style="width:70%">
     </div>
     <div class="fragment fade-in-then-out" data-fragment-index="5" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_down.png" style="width:70%">
     </div>
     <div class="fragment fade-in-then-out" data-fragment-index="7" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_ranked.png">
     </div>
 -->        <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<!--<section id='example_data_pinto'>
    <h1>Example Data - Pinto et al. SARS-CoV-2 study</h1>
    <p>The <a href="https://www.cell.com/iscience/fulltext/S2589-0042(22)02168-X" target="_blank">Pinto et al</a> study is a multi-OMICs study of SARS-CoV-2 host responses in lung epithelial cells. The data files below were adapted from the supplementary data files provided with the publication, which were already pre-filtered.</p>
    <ol>
        <li class="fragment" data-fragment-index="1">Differentially regulated transcripts identified in cells infected with SARS-CoV2; <highlight>list of 2656 genes (NCBI gene symbol)</highlight>. <a href="Pinto_PMID36590899_mmc2_transcript_ncbi.txt">Download data</a>.</li>
        <li class="fragment" data-fragment-index="3">Differentially expressed proteins identified in cells infected with SARS-CoV2; <highlight>list of 225 proteins</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/Pinto_PMID36590899_mmc3_protein_uni.txt">Download data with Uniprot identifiers</a> / <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/Pinto_PMID36590899_mmc3_protein_ncbi.txt">Download data with NCBI gene symbols</a>.</li>
    </ol>
    <p class="fragment" data-fragment-index="5">These input files can be used for <b>ORA</b> analysis in <b>Enrichr</b> and <b>WebGestalt</b>.</p>
     <div class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_diff_transcript.png" style="width:60%">
     </div>
    <div class="fragment fade-in-then-out" data-fragment-index="4" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_diff_protein.png" style="width:70%">
     </div>
</section>-->
                
<section id='example_data_pinto'>
    <h1>Example Data - Pinto et al. SARS-CoV-2 study</h1>
    <p>The <a href="https://www.cell.com/iscience/fulltext/S2589-0042(22)02168-X" target="_blank">Pinto et al</a> study is a multi-OMICs study of SARS-CoV-2 host responses in lung epithelial cells. The data files below were adapted from the supplementary data files provided with the publication, which were already pre-filtered.</p>
    <ol>
        <li>Differentially regulated transcripts identified in cells infected with SARS-CoV2; <highlight>list of 2656 genes (NCBI gene symbol)</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/Pinto_PMID36590899_mmc2_transcript_ncbi.txt" target="_blank">Download data</a>.</li>
        <li>Differentially expressed proteins identified in cells infected with SARS-CoV2; <highlight>list of 225 proteins</highlight>. <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/Pinto_PMID36590899_mmc3_protein_uni.txt" target="_blank">Download data with Uniprot identifiers</a> / <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/Pinto_PMID36590899_mmc3_protein_ncbi.txt" target="_blank">Download data with NCBI gene symbols</a>.</li>
    </ol>
    <p>These input files can be used for <b>ORA</b> analysis in <b>Enrichr</b> and <b>WebGestalt</b>.</p>
     <!--<div class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_diff_transcript.png" style="width:60%">
     </div>
    <div class="fragment fade-in-then-out" data-fragment-index="4" style="position: absolute; left: 380px; top: 180px">
            <img  src="input_files_diff_protein.png" style="width:70%">
     </div>-->
</section>

<!--<section id='example_data_voineagu'>
         <h1>Example Data - Voineagu et al. autism study</h1>
    <p>The <a href="https://europepmc.org/article/MED/21614001" target="_blank">Voineagu et al.</a> study compares the transcriptome between autistic and normal brain, and was downloaded from <a href="https://www.ebi.ac.uk/gxa/experiments-content/E-GEOD-30573/resources/ExperimentDownloadSupplier.BulkDifferential/tsv" target="_blank">Expression Atlas</a>. It contains data for all genes measured with the following comma-separated columns:</p>
        <ul>
            <li>NCBI gene symbol</li>
            <li>fold change</li>
            <li>p value</li> 
       </ul>
    <p><a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/E-GEOD-30573.csv">Download data</a>.</p>
    <div class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 50px; top: 120px">
            <img  src="input_files_csv.png" style="width:60%">
     </div>
    <p class="fragment" data-fragment-index="3">This input is used for <b>ORA</b> and <b>GSEA</b> analysis in the <highlight>Interactive Enrichment Analysis</highlight> tool (data is pre-loaded in the tool).</p>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>-->
                
<section id='example_data_voineagu'>
         <h1>Example Data - Voineagu et al. autism study</h1>
    <p>The <a href="https://europepmc.org/article/MED/21614001" target="_blank">Voineagu et al.</a> study compares the transcriptome between autistic and normal brain, and was downloaded from <a href="https://www.ebi.ac.uk/gxa/experiments-content/E-GEOD-30573/resources/ExperimentDownloadSupplier.BulkDifferential/tsv" target="_blank">Expression Atlas</a>. It contains data for all genes measured with the following comma-separated columns:</p>
        <ul>
            <li>NCBI gene symbol</li>
            <li>fold change</li>
            <li>p value</li> 
       </ul>
    <p><a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/E-GEOD-30573.csv" target="_blank">Download data</a>.</p>
    <!--<div class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 50px; top: 120px">
            <img  src="input_files_csv.png" style="width:60%">
     </div>-->
    <p>This input is used for <b>ORA</b> and <b>GSEA</b> analysis in the <highlight>Interactive Enrichment Analysis</highlight> tool.</p>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='navigation 1'>
<a href="#/enrichr" class="navigate-next"><ribbon>Enrichr</ribbon></a>
<a href="#/webgestalt"><ribbon>WebGestalt</ribbon></a>
<a href="#/iea"><ribbon>Interactive Enrichment Analysis</ribbon></a>
<a href="#/downstream_analysis"><ribbon>Downstream Analysis Options</ribbon></a>

<aside class="notes" data-markdown>
Add speaker notes here...
</aside>
</section>

<section id='enrichr'>
    <h1>Enrichr</h1>
         <p>The <a href="https://maayanlab.cloud/Enrichr/" target="_blank">Enrichr</a> tool offers an easy-to-use interface for basic Over-Representation Analysis (ORA) for a large number of gene set libraries.</p>
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 50px; top: 200px">
            <img class="plain" src="enrichr.png" style="width:70%">
        </div>
        <div class="fragment" data-fragment-index="2">
             <p>Before starting analysis, you can browse the available gene sets under <highlight>Libraries</highlight> in the top menu.</p>
            <img class="plain" src="enrichr_lib.png" style="width:70%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_libraries'>
    <h1>Enrichr: Gene Set Libraries</h1>
         <p>Enrichr includes over <b>200</b> gene set libraries, including Gene Ontology terms, pathways, disease-associated sets, cell type markers etc.</p>
         <p class="fragment" >You can also search for gene sets either by the term name under <highlight>Term search</highlight> or by gene under <highlight>Gene search</highlight>.</p>
         <img class="plain" src="enrichr_libraries.png" style="width:45%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_start'>
    <h1>Enrichr: Start Analysis</h1>
         <p>Analysis is started by simply copying the list of genes from an input text file into the input box on the right of the <highlight>Analyze</highlight> tab and clicking <highlight>Submit</highlight>.</p>
         <p class="fragment">Note that this corresponds to the TCGA example data file containing <a href="https://raw.githubusercontent.com/gladstone-institutes/Bioinformatics-Workshops/master/intro-pathway-analysis/ExampleData/lung.expr.up.txt" target="_blank">up-regulated genes</a>.</p>
         <img class="plain" src="enrichr_start.png" style="width:55%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_results'>
    <h1>Enrichr: Results Overview</h1>
         <p>The results are displayed as a grid of libraries for each functional category (at the top), the image below shows the results in the <highlight>Pathways</highlight> category.</p>
         <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; top: 200px; left: 60px">
         <img class="plain" src="enrichr_results.png" style="width:55%"></div>
         <div class="fragment fade-in" data-fragment-index="2">
         <p class="fragment" data-fragment-index="2">Clicking on one of the squares will show the detailed results for that library:</p>
             <img class="plain" src="results_wp.png" style="width:50%">
         </div>         

         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_details'>
    <h1>Enrichr: Gene Set Results</h1>
         <p>Looking at the results for a specific library in detail, the gene sets are sorted by p-value ranking by default. Clicking on any of the bars representing gene sets re-sorts the bar graph by the different score.</p>
         <div
	     style="position: absolute; top: 215px; left: 160px; width: 80%" >
		<img class="plain"
		     src="results_wp_detail.png">
	</div>
    	<div class="fragment fade-in" data-fragment-index="2"
	     style="position: absolute; top: 220px; left: 160px; width: 82%" >
		<img class="plain fragment" data-fragment-index="2"
		     src="results_wp_sorted.png">
	</div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_save'>
    <h1>Enrichr: Save Results</h1>
         <p>To download an image of the bar chart, click either of the <highlight>SVG PNG JPG</highlight> buttons at the top right of the chart. </p>
         <img class="plain" src="results_wp_save.png" style="width:70%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='enrichr_table'>
    <h1>Enrichr: Table View</h1>
         <p>In addition to the default bar chart, results are also available as a table under <highlight>Table</highlight>. The table include the four enrichment scores calculated by Enrichr: <highlight>P-value</highlight>, <highlight>Adjusted p-value</highlight>, <highlight>Odds Ratio</highlight> and <highlight>Combined score</highlight>. More information about these scores can be found in the <a href="https://maayanlab.cloud/Enrichr/help#background&q=4" target="_blank">Enrichr documentation</a>.</p>
         <p class="fragment" data-fragment-index="2">At the bottom of the table there is a link to <highlight>Export entries to table</highlight> which will download a .txt file with the full results table.</p>
         <img class="plain" src="results_wp_table.png" style="width:68%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='navigation 2'>
<a href="#/enrichr"><ribbon>Enrichr</ribbon></a>
<a href="#/webgestalt" class="navigate-next"><ribbon>WebGestalt</ribbon></a>
<a href="#/iea"><ribbon>Interactive Enrichment Analysis</ribbon></a>
<a href="#/downstream_analysis"><ribbon>Downstream Analysis Options</ribbon></a>

<aside class="notes" data-markdown>
Add speaker notes here...
</aside>
</section>
                
<section id='webgestalt'>
    <h1>WebGestalt</h1>
         <p><a href="http://www.webgestalt.org/" target="_blank">WebGestalt</a> offers more advanced analysis options for a smaller number of gene set libraries. Using WebGestalt, it is possible to run either Over-Representation Analysis (ORA), Gene Set Enrichment Analysis (GSEA) or Network Topology-based Analysis (TNA).</p>
         <img class="plain" src="webgestalt.png" style="width:80%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='webgestalt_datasource'>
    <h1>WebGestalt: Data source</h1>
	<p>WebGestalt data sources are listed under <highlight>Data source</highlight> and includes Gene Ontology terms, pathways, networks, disease-associated sets, drug terms etc.</p>
         <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 200px; top: 200px">
           <img class="plain" src="webgestalt_datasource.png" style="width:35%">
         </div>
        <div class="fragment fade-in" data-fragment-index="2" style="position: absolute; left: 200px; top: 200px">
         <img class="plain" src="webgestalt_datasource2.png" style="width:35%">
    </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='webgestalt_start1'>
    <h1>WebGestalt: Start Analysis</h1>
         <p>To start analysis, the <highlight>Basic parameters</highlight> section should be completed with the correct <highlight>Organism of interest</highlight>, <highlight>Method of Interest</highlight> (ORA, GSEA or NTA) and <highlight>Functional Database</highlight>. Choose a functional database category, for example <b>pathway</b>, and then specify which data source to use, for example <b>Wikipathway</b>.</p>
         <img class="plain" src="webgestalt_basic.png" style="width:70%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='webgestalt_start2'>
    <h1>WebGestalt: Start Analysis</h1>
         <p>The input data is defined in the <highlight>Gene List</highlight> section, and the dataset or list of genes can either be pasted into the input field or uploaded as a text file. <highlight>Select Gene ID Type</highlight> should match the identifier type in the input data. The list used here corresponds to the TCGA example input file containing the full dataset with associated gene rank.</p>
         <p class="fragment">Clicking <highlight>Submit</highlight> at the bottom left starts the analysis.</p>
         <img class="plain" src="webgestalt_inputdata.png" style="width:70%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='webgestalt_results_summary'>
    <h1>WebGestalt: Results Summary</h1>
         <p>The top of the results page includes a job summary and a link to download the full results.</p>
         <img class="plain" src="webgestalt_summary.png" style="width:90%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='webgestalt_barchart'>
    <h1>WebGestalt: Enrichment Results</h1>
         <p>Results are displayed as a bar chart by default. Right-clicking on the bar chart lets you download in either <highlight>PNG</highlight> or <highlight>SVG</highlight> format.</p>
         <p class="fragment" data-fragment-index="2">Clicking on the bars updates the pathway-specific display at the bottom of the page.</p>
         <img class="plain" src="webgestalt_barchart.png" style="width:70%">
    	<div class="fragment fade-in" data-fragment-index="3"
	     style="position: absolute; top: 250px; left: 175px; width: 80%" >
		<img class="plain fragment" data-fragment-index="3"
		     src="webgestalt_pathwayspecific.png">
	</div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='webgestalt_enrichmentplot'>
    <h1>WebGestalt: Enrichment Results</h1>
         <p>The results specific to a gene set (pathway in this case) include the scoring statistics calculated for the enrichment, including the enrichment score, a sortable table and the enrichment plot.</p>
         <p class="fragment" data-fragment-index="2">The enrichment plot is described in detail <a href="https://www.gsea-msigdb.org/gsea/doc/GSEAUserGuideFrame.html" target="_blank">here</a>. Briefly, the plot 
		 in the upper half represents the running enrichment score (ES), as the analysis walks down the ranked list of genes (bottom section), starting at the most highly ranked gene. When a gene in the pathway is in 
		 the ranked list, the score goes up, and when it’s not it goes down. The corresponding overlap between the ranked list and the gene set is indicated by a line in the middle section. The top score in the enrichment 
		 map plot is the score reported for the particular gene set.</p>    
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 50px; top: 250px">
            <img src="webgestalt_pathwayspecific.png" class="plain">
        </div>
        <div data-fragment-index="2" class="fragment fade-in" style="position: absolute; left: 240px; top: 420px">
        <img data-fragment-index="2" src="enrichment_score_plot.png" class="plain fragment" width="70%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='webgestalt_pathway'>
    <h1>WebGestalt: Enrichment Results</h1>
         <p>For WikiPathways results, the link in the upper left is clickable and will open a pathway view with the overlapping/leading edge genes highlighted.</p>         
        <img class="plain" src="wp_rbgenes.png" style="width:70%">
    
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='webgestalt_other'>
    <h1>WebGestalt: Enrichment Results</h1>
        <p>In addition to the bar chart, the main results overview can also be visualized as a table or volcano plot by clicking the <highlight>Table</highlight> and <highlight>Volcano</highlight> buttons at the top left.</p>
        <p class="fragment" data-fragment-index="2">The volcano plot has options for customized downloads available, as well as pan/zoom controls.</p>  
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; top: 200px">
        <img  src="webgestalt_table.png" class="plain">
        </div>
        <div data-fragment-index="2" class="fragment fade-in" style="position: absolute; top: 250px">
        <img src="webgestalt_volcano.png" class="plain" style="width:70%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='navigation 3'>
<a href="#/enrichr"><ribbon>Enrichr</ribbon></a>
<a href="#/webgestalt"><ribbon>WebGestalt</ribbon></a>
<a href="#/iea" class="navigate-next"><ribbon>Interactive Enrichment Analysis</ribbon></a>
<a href="#/downstream_analysis"><ribbon>Downstream Analysis Options</ribbon></a>

<aside class="notes" data-markdown>
Add speaker notes here...
</aside>
</section>
                
<section id='iea'>
    <h1>Interactive Enrichment Analysis</h1>
        <p><a href="https://github.com/gladstone-institutes/Interactive-Enrichment-Analysis" target="_blank">Interactive Enrichment Analysis</a> is a user-friendly interactive tool to perform enrichment analysis for multiple datasets across multiple public databases. This tool can run both Over-Representation Analysis(ORA) and Gene Set Enrichment Analysis (GSEA).</p>
        <img class="plain" src="interactive_enrichment_analysis.png" style="width:80%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
             
<section id='iea_database'>
    <h1>Interactive Enrichment Analysis: Database</h1>
        <p>A set of public database collections is provided in the drop-down in the <highlight>Database</highlight> section. These are updated monthly and date stamped. It is also possible to construct your own database collection under <highlight>BUILD NEW DATABASE</highlight>, and providing one or more <a href="https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29" target="_blank">GMT</a> files.</p>
        <p class="fragment" data-fragment-index="2">Once a database collection is selected, the specific databases contained in the collection will be displayed.</p>
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 0px; top: 250px">
        <img  src="iea_database.png" class="plain" style="width:80%">
        </div>
        <div data-fragment-index="2" class="fragment fade-in" style="position: absolute; left: 40px; top: 330px">
        <img src="iea_database_contents.png" class="plain" style="width:60%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>    
                
<section id='iea_dataset1'>
    <h1>Interactive Enrichment Analysis: Datasets</h1>
        <p>Datasets are selected in the <highlight>Datasets</highlight> section; one or more datasets can be selected if they are all of the same format. A column named <b>gene</b> is required, and additional columns named <b>rank</b>, <b>p.value</b>, and <b>fold.change</b> are allowed and necessary for GSEA.</p>
        <img class="plain" src="iea_select_dataset.png" style="width:60%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
  
<section id='iea_dataset2'>
    <h1>Interactive Enrichment Analysis: Dataset Options</h1>
        <p>The tool will display a preview of the first few rows of the first chosen dataset along with the required and optional columns that were detected.</p>
        <p class="fragment" data-fragment-index="2"><highlight>Gene identifier</highlight> and <highlight>organism</highlight> should match the data. The analysis parameters can also be adjusted here.</p>  
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 0px; top: 200px">
        <img src="iea_dataset_options.png" class="plain"style="width:50%">
        </div>
        <div data-fragment-index="2" class="fragment fade-in" style="position: absolute; left: 0px; top: 270px">
        <img src="iea_dataset_options_highlight.png" class="plain" style="width:45%">
        </div>
        <img class="plain" src="" style="width:70%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_start'>
    <h1>Interactive Enrichment Analysis: Start Analysis</h1>
        <p>When <highlight>Databases</highlight> and <highlight>Datasets</highlight> are loaded and verified, the side panel will be updated with a <highlight>Run Analyses</highlight> option.</p>
        <p class="fragment" data-fragment-index="2">Once analysis is started, the initial set of panels will collapse and a <highlight>Progress</highlight> panel will open along with a progress bar in the lower-right corner. The <highlight>Progress</highlight> panel will report the individual analysis steps, any errors, as well as start and finish times for the run, and the location of the output folder.</p>  
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 0px; top: 200px">
        <img src="iea_startanalysis.png" class="plain"style="width:80%">
        </div>
        <div class="fragment fade-in" data-fragment-index="2" style="position: absolute; left: 0px; top: 350px">
        <img src="iea_progress.png" class="plain"style="width:60%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_viewresults'>
    <h1>Interactive Enrichment Analysis: View Results</h1>
        <p>Analysis results can be viewed by clicking <highlight>View Results</highlight>. Results are viewed in another app that allows you to interactively explore results and download individual files.</p>
        <p class="fragment" data-fragment-index="2">In the results app, the <highlight>Data</highlight> tab displays the processed input data as a searchable table and volcano plot.
        A checkbox at the top allows you to also view the genes that were excluded from the analysis, e.g., due to failed identifier mapping. Both the table and plot are downloadable.</p>
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 0px; top: 250px">
        <img src="iea_progress_viewresults.png" class="plain"style="width:60%">
        </div>
        <div class="fragment fade-in" data-fragment-index="2" style="position: absolute; left: 0px; top: 320px">
        <img src="iea_results_data_ora.png" class="plain" style="width:70%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_dataplots'>
    <h1>Interactive Enrichment Analysis: Data plots</h1>    
        <p class="fragment fade-out" data-fragment-index="1" style="position: absolute; left: 0px; top: 100px">The input data is visualized in a volcano plot, which plots genes by statistical significance, <b>p.value</b>, versus magnitude of change, <b>fold.change</b>. Genes are highlighted based on the p.value and fold-change cutoffs selected during setup. Selected genes are labeled; select genes either by top <b>n</b> genes, or by selection by name.</p>  
        <p class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 0px; top: 100px">A bar plot of your input data is also available (via a drop-down), highlighting positive and negative fold-change values for a subset of genes, either top <b>n</b> genes or genes selected by name.</p>  
        <div data-fragment-index="1" class="fragment fade-out" style="position: absolute; left: 220px; top: 270px">
            <img class="plain" src="shiny_results_data_volcano.png" >
        </div>
        <div data-fragment-index="2" class="fragment fade-in-then-out" style="position: absolute; left: 220px; top: 200px">
            <img class="plain" src="shiny_results_data_bar.png">
        </div>
    
    <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results1'>
    <h1>Interactive Enrichment Analysis: Analysis Results</h1>
        <p>The <highlight>Results</highlight> tab displays a table of enriched terms/pathways, a plot of top results, and a plot of a selected result from the table. The table is paginated and supports selection, search and filtering. The plots for top results include <b>dot</b>, <b>heatmap</b>, <b>emap</b>, <b>concept networks</b>, etc.</p>
        <p class="fragment" data-fragment-index="2">The table and plots can all be downloaded.</p>  
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left:0px; top: 230px">
        <img  src="iea_results_ora.png" class="plain" style="width:60%">
        </div>
        <div class="fragment fragment fade-in" data-fragment-index="2" style="position: absolute; left: 0px; top: 300px">
        <img src="iea_results_ora_download.png" class="plain" style="float:left; width:48%">
        <img src="iea_results_dot_download.png" class="plain" style="float:right; width:48%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results2'>
    <h1>Interactive Enrichment Analysis: Analysis Results</h1>
        <p>Using the left side panel, one can navigate between GSEA and ORA methods while viewing the results for a particular database to explore hits in common (and unique) to the different methods. Here we are comparing the <b>ORA</b> and <b>GSEA</b> results for the WikiPathways database, note the differences in the table and dot plot.</p> 
        <div style="position: absolute; left:0px; top: 260px">
        <img src="iea_results_ora_pw.png" class="plain" style="float:left; width:48%">
        <img src="iea_results_gsea_pw.png" class="plain" style="float:right; width:48%">
    </div>
            <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results_dot'>
    <h1>Interactive Enrichment Analysis: Dot plot</h1>
        <p>A <highlight>dot plot</highlight> of results highlight multiple statistics for the top enriched terms/pathways. By default, the size of dots corresponds to the <b>percentage</b> of dataset genes overlapping a term/pathway versus all genes in the term/pathway, and the color represents <b>adjusted p value</b>. These parameters and other plot view options are customizable in a set of controls under the plot.</p>
        <div style="position: absolute; left: 250px; top: 250px">
        <img src="shiny_results_res_dot.png" class="plain" style="width:75%">
        </div>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results_overlap'>
    <h1>Interactive Enrichment Analysis: Overlapping genes</h1>
        <p class="fragment fade-out" data-fragment-index="1" style="position: absolute; left:  100px; top: 100px">Several plot types are available for visualizing the gene overlap between results.</p>
        <div class="fragment fade-in-then-out" data-fragment-index="2" style="position: absolute; left: 0px; top: 100px">
        <p><highlight>Heatmap</highlight> plots the top results (terms/pathways) versus top genes to highlight overlapping genes among the most enriched results. As with all plots, the view options are customizable.</p>  
            <img class="plain" src="shiny_results_res_heatmap.png" style="width:60%">
        </div>
        <div class="fragment fade-in-then-out" data-fragment-index="3" style="position: absolute; left: 0px; top: 100px">
        <p><highlight>Enrichment map</highlight> (emap) connects results (terms/pathways) with overlapping genes in a network view. The interactions (edges) in the network represent gene overlap between terms/pathways (nodes), and nodes are sized based on the number of genes in them.</p> 
            <img class="plain" src="shiny_results_res_emap.png" style="width:40%">
        </div>
        <div class="fragment fade-in" data-fragment-index="4" style="position: absolute; left: 0px; top: 100px">
        <p><highlight>Concept network</highlight> (cnet) connects results via their overlapping genes. Results (terms/pathways) are represented by the larger nodes, and each gene in that term/pathway is connected via an interactions to it. Interactions are color-coded by term/pathway. In this way, it is easy to see which genes are relevant to more than one term/pathway.</p>
             <img class="plain" src="iea_results_cnet.png" style="width:50%">
        </div>
     <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results_gseaplot'>
    <h1>Interactive Enrichment Analysis: GSEA plot</h1>
        <p>For each result in the results table, results-specific plots are available based on the database the results is from and the analysis method used. For example, for <b>GSEA</b> analysis, the <b>GSEA Enrichment Score</b> plot is available. As described earlier, the plot in the upper half represents the running enrichment score (ES), as the analysis walks down the ranked list of genes. The middle section shows where the members of the gene set appear in the ranked list of genes. The top score in the enrichment map plot is the score reported for the particular gene set.
        </p>
        <img src="iea_ES_plot.png" class="plain" style="width:55%">
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
<section id='iea_results_wpplot'>
    <h1>Interactive Enrichment Analysis: Pathway plot</h1>
        <div class="fragment fade-out" data-fragment-index="1" style="position: absolute; left:  100px; top: 100px">
        <p>Some database-specific visualizations are also available, for example <b>WikiPathways</b> visualizations of the data on models from the <a href="https://www.wikipathways.org/" target="_blank">WikiPathways</a> database. Genes that met the criteria for fold change and p value cutoff are highlighted in the pathway (orange for up, blue for down). The <b>WikiPathways</b> button will take you to the pathway model at WikiPathways.</p>
        <img src="iea_pathway_plot.png" class="plain" style="width:55%">
        </div>
        <div class="fragment fade-in" data-fragment-index="2" style="position: absolute; left: 0px; top: 100px">
        <p>Opening the pathway in a new window gives you access to a larger, interactive format of the pathway.</p>   
            <img src="iea_pathway_plot_large.png" class="plain" style="width:90%">
        </div>
        
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>

<section id='downstream_analysis'>
    <h1>Functional Enrichment Analysis: Downstream analysis</h1>
        <p>Once you have completed exploring your data and results in either of these tools, there are several options for continued analysis and exploration:</p>
        <ul>
            <li>Create custom <a href="https://cytoscape.org/cytoscape-tutorials/protocols/wikipathways-app/#/title" target="_blank">pathway visualizations of your data in Cytoscape</a>.</li>
            <li>Explore network visualization and analysis in <a href="https://tutorials.cytoscape.org" target="_blank">Cytoscape</a>.</li>
            <li>Extend pathways with regulatory interactions using <a href="https://cytargetlinker.github.io/pages/tutorials" target="_blank">CyTargetLinker</a>.</li>
            <li>Create pathway models relevant to your research, or modify existing models, with <a href="https://github.com/gladstone-institutes/Bioinformatics-Workshops/wiki/Introduction-to-Pathway-Modeling" target="_blank">PathVisio and WikiPathways</a>.</li>
    
    </ul>
         <aside class="notes" data-markdown>
         Notes go here
         </aside>
</section>
                
                
<!-- THIS IS WHERE YOUR CONTENT ENDS -->
<!-- ========================================================== -->

			</div>
		</div>

		<script src="../../lib/js/head.min.js"></script>
		<script src="../../js/reveal.js"></script>

<script>
	Reveal.addEventListener( 'ready', function() {
	if(!window.navigator.userAgent.includes('CyBrowser')){
		var divs = document.getElementsByClassName("cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	} else {
		var divs = document.getElementsByClassName("not-cybrowser")
		for (var i=0;i<divs.length;i++){
			divs[i].style.display = "none";
		}
	}
});
</script>

<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-155159-15', 'auto');
  ga('send', 'pageview');

</script>
		<script>
			// More info https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				history: true,
				controls: true,
				progress: true,
				slideNumber: false,
				touch: true,
				fragments: true,
				help: true,
				showNotes: false,
				previewLinks: false,
				transition: 'slide', // none/fade/slide/convex/concave/zoom
				width: 1200,
				height: 860,
				autoSlide: 0,

				// More info https://github.com/hakimel/reveal.js#dependencies
				dependencies: [
				{ src: '../../plugin/markdown/marked.js' },
				{ src: '../../plugin/markdown/markdown.js' },

				{ src: '../../plugin/zoom-js/zoom.js', async: true, condition: function(){ return !!document.body.classList; } },
				{ src: '../../plugin/notes/notes.js', async: true },
				{ src: '../../plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: '../../plugin/external/external.js', condition: function() { return !!document.querySelector( '[data-external]' ); } }
				]
				});
			</script>
		</body>
	</html>
